e2eAIOK/ModelAdapter/readme.md

```diff
@@ -0,0 +1,178 @@
+
+Training pipeline with Model Adaptor is shown as following. The colored boxes are Model Adaptor-specific, which is the difference between pipeline with Model Adaptor and the general pipeline. 
+
+**Model Adaptor-specific components**:
+- **Pre-trained model**: A pre-trained model is served as a knowledge base to Model Adaptor. Model Adaptor transfer knowledge from pre-trained models to target task.
+  Model Adaptor provides many built-in pre-trained models by default, however, users can specify their own pre-trained model.
+- **Source Domain data**: Source domain data, which is much bigger than target data,  is served as another knowledge base to Model Adaptor. Model Adaptor transfer knowledge from source domain data to target task. 
+  
+- **Unified API**: Unified API is a unified interface to assign different transfer learning ability to the underlying model. The Unified API returns a transferable model, which is a wrapper of the underlying model. The type of transferable model is TransferableModel, which shares the same interface with the underlying model.
+- **Finetuner**: A finetuner is one of the components for TransferableModel, performs finetuning and servs as a regularizer of underlying model. 
+- **Distiller**: A distiller is second of the components for TransferableModel, performs distillation and servs as another regularizer of underlying model. 
+- **Adapter**: : An adapter is the third of the components for TransferableModel, performs domain adaption and serves as third regularizer of underlying model.
+
+**General components**:
+- **Target data**: Target data defines the target task, and the trained model should be well predictive on unseen target data (called "test dataset"). Target data maybe few labeled or even unlabeled, which is a big challenge, as mentioned above.
+- **Data Preprocessor**: Data preprocessor preprocesses raw data, i.e. text or image, to formatted data to feed into model.
+- **Train Dataset/Validate Dataset**: Formatted data generated by data preprocessor is split into train dataset and validate dataset. Train dataset is used to train model, and Validate dataset is used to tune hyper-parameters.
+- **Optimizer**: Optimizer defines how to optimize model parameters step-by-step, i.e. Stochastic Gradient Descent (SGD).
+- **Scheduler**: Scheduler defines how to schedule learning rate step-by-step, i.e. StepLR which decays learning periodically.
+
+<img src="./doc/imgs/arch.png" width="800px" />
+
+## Getting Started 
+### Installing
+1. Option 1: install with Docker (expected in v0.4)
+   ```bash
+   git clone https://github.com/intel/e2eAIOK.git
+   cd e2eAIOK; python scripts/start_e2eaiok_docker.py --backend  pytorch --dataset_path ./ --workers host1, host2, host3, host4
```

Comment:
@XinyaoWa , do you use pytorch as backend instead of pytorch112? Does this install guide verified ? 
This is how DeNas enabled: https://github.com/intel/e2eAIOK/tree/main/e2eAIOK/DeNas

Reply:
Meanwhile, I remembered you have some minor change to Dockerfile, didn't see in this PR

Reply:
Readme and Dockerfile updated.